go.mod

```diff
@@ -44,18 +44,15 @@ require (
 	github.com/tsenart/vegeta v12.7.0+incompatible
 	go.uber.org/zap v1.12.0
 	golang.org/x/crypto v0.0.0-20191011191535-87dc89f01550
-	golang.org/x/net v0.0.0-20191011234655-491137f69257 // indirect
 	golang.org/x/sys v0.0.0-20191010194322-b09406accb47 // indirect
 	golang.org/x/time v0.0.0-20190921001708-c4c64cad1fd0 // indirect
-	golang.org/x/xerrors v0.0.0-20191011141410-1b5146add898 // indirect
 	google.golang.org/appengine v1.6.5 // indirect
 	gopkg.in/yaml.v2 v2.2.4
 	gopkg.in/yaml.v3 v3.0.0-20191120175047-4206685974f2
-	k8s.io/api v0.0.0-20191114100352-16d7abae0d2a
-	k8s.io/apimachinery v0.0.0-20191028221656-72ed19daf4bb
+	k8s.io/api v0.17.1
+	k8s.io/apimachinery v0.17.1
```

Comment:
Did you bump those on purpose? I think we use the ones coming with kubebuilder/controller-runtime by default.

Reply:
I did, or I should say: The ocp dependency did. Do you think this will be a problem?

Reply:
I think in theory that's a problem.
But in practice that's probably not 😄 I would be surprised if that causes actual troubles.

However, the right way to deal with this IMO is to make the `deployer` have its own `go.mod` and `go.sum`, entirely decoupled from the operator.
I think there's no reason for both projects to share their dependencies (which does not mean they cannot live in the same github repo).

Reply:
I think I would lean on the safe side and keep these dependencies driven by the controller runtime if it's possible. It is already the case for the license detector so I guess we can do the same for the deployer.

Reply:
Unfortunately, I think not bumping the version may be an issue for this patch. `openshift/api` depends on these versions so I'll probably take @sebgl suggestion and add a `go.mod` and `go.sum` for the `deployer`.

Any thoughts/opinions on how you would prefer to have this done?

Reply:
I should note that this is a requirement for the `tests` only, which means we could add these module files to the `test/e2e` only.

Reply:
> I think we can avoid to rely on the OCP client if we just need to patch the SCC , which is a CRD starting 4.x btw:
> 
> ```
> $ k get crd securitycontextconstraints.security.openshift.io 
> NAME                                               CREATED AT
> securitycontextconstraints.security.openshift.io   2020-02-07T06:52:43Z
> ```
> 
> This can be done by using the legacy K8S client, using a `PATCH` request on the right endpoint:
> 
> ```go
> user := fmt.Sprintf(`"system:serviceaccount:%s:%s"`, 
>                            b.ServiceAccount.Namespace,
>                            b.ServiceAccount.Name,
>                  )
> // The patch below adds the service account user in the 'users' fields of a SCC
> patch := []byte(`{ "users": [` + user + `]}`)
> 
> // We want to patch the anyuid SCC. In term of url it means that we need to send a patch request to:
> // https://<Openshift URL>/apis/security.openshift.io/v1/securitycontextconstraints/anyuid
> patchClient := k8sClient.RESTClient().
>     Patch(types.MergePatchType).
>     Prefix("apis", "security.openshift.io", "v1").
>     Resource("securitycontextconstraints").
>     Name("anyuid").
>     Body(patch)
> 
> result := patchClient.Do()
> ```
> 
> See a full patch [here](https://github.com/elastic/cloud-on-k8s/compare/openshift...barkbay:openshift-patch-with-native-client#diff-b7b20c2a1a27a46b32159937c7485845R33)
> 
> It would avoid to rely on those dependencies and deal with new go module files. On other benefit is that it seems to work with both OCP 4.x and 3.11, I'm not sure if the official OCP client for 4.x is compatible with 3.11.

Oh, nice! Thanks. I had tried something like this but I think I did something wrong 'cause it didn't work for me. I'll go with this just to unblock this PR. 

As a side note, it feels a bit hacky in the sense that we are avoiding a, perhaps more important, issue that is handling SCCs. I understand we are just talking about our test suite here and not the operator itself, however, I think it would be better to deal with this properly by using the actual clients.

Thank you for taking the time to write this patch

Reply:
@barkbay patch applied perfectly. Just launched tests now, will report back soon :)